### DISCUSSION TITLE
Welcome to arr-scripts Discussions!

### DISCUSSION THREAD

**Discussion by RandomNinjaAtk** [RandomNinjaAtk/arr-scripts#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Allram** [comment]
Thank you for making these awesome scripts for the *arrs!
I was using your docker containers, when i saw that you have changed to script based solution instead, and i see your point on this. And this was quite quick to set up and maybe more "future-proof", as we now can use "official"-containers from Lsio.

Again, Thank you for the awesome work you do.

**Reply by RandomNinjaAtk** [reply]
You can also now use any tag you want, go nightly, stable or develop... However, if your keeping your existing app data to move over, you have to use develop  or nightly tag because the original containers were based on develop...

Edit: Sonarr has to be V4 (develop)

**Reply by KINGLIFER** [reply]
How the heck do you install it? Why is everything so vague. Like seriously. Tried the lidarr one... wow. did nothing. 
I changed permissions to root... still did not download any scripts.
why brother... why. LOL

**Reply by RandomNinjaAtk** [reply]
Did you follow the instructions in the readme….?

it’s pretty simple, you download one file and put it in the correct folder… as outlined in the instructions…

**Reply by KINGLIFER** [reply]
I did. Apparently I had to download two files. scripts_init.bash and extended.conf

**Reply by RandomNinjaAtk** [reply]
@KINGLIFER 

You should open your own discussion, because if you followed the directions correctly, it just works…

So open your thread and be detailed about what you did… post logs, docker compose config or run commands and etc… 

Making a statement without any diagnostic info, won’t enable anyone to help you. 

Edit: extended.conf is automatically downloaded for you. Only the init script is needed

**Reply by coromjones97** [reply]
I am having the same issues

**Reply by stuartmhannon** [reply]
@RandomNinjaAtk  - I appreciate your work on this, but have read over your instructions, followed to the letter, and I'm getting a similar result as others have posted. Your reoccurring "follow the instructions, it just works" - is not helpful at all. There are no real instructions on here to tell new users HOW to use these scripts. You keep saying "they just work!" when clearly many users do not have that experience. Developing software is partly about coding and partly how humans interact with your code. You may need to reevaluate how you present your project to the world.

**Reply by RandomNinjaAtk** [reply]
If your having an issue, please feel free to open an issue.

Provide the steps to reproduce and logs to provide clarity. Every instance that I've seen a user have an issue has always resulted in user-error...  Nothing can be done to resolve an issue if it cannot be reproduced and is not documented properly....

The code/scripts here are as-is... As stated in the readme. I generally try to assist where possible to an extent that time allows... Feel free to create or use your own solution if you want....

Additionally, feel free to contribute if you have ways of improving it... Which could be as simple as improving the documentation...

Edit:

Generally these scripts work in the background and are designed to work without any user intervention/action. Which is why there isn't much in the way of information on usage, because your not supposed to do anything... You configure the config file, and the scripts just do their thing automatically, which is the entire point... Viewing script/console logs is the only real way to see what is happening, unless you disabled autorun and are running the scripts manually... But in that case, you chose to do that...

**Reply by WillScott73** [reply]
I do get Stuart's frustration, and as a developer myself I recognise some of his observations of developers, but one would think that someone using GitHub would have some basic knowledge of troubleshooting.

I wonder if he's having the issue that I had, and that's of needing to change the EOL characters if you check out to Windows and then copy the files over to Linux? That's not a code issue!!

**Reply by KINGLIFER** [reply]
There are numerous factors contributing to the challenges faced in dealing with this issue. These encompass (and may be one or more) reasons like personal life, obligatory manipulation, time constraints, a superiority complex, language barriers, imposter syndrome, hidden complexities, fear, immaturity, inadequate user experience skills, sheer indifference, and a failure to employ various setups or scenarios before deploying the software to the public. It's evident that the documentation intentionally remains vague, raising the crucial question of why, though not serving as a direct solution to the problem.

We are in an age where we talk like how the "old timers" used to talk... It is no longer "it may need to warm up" or "walk it off" now it is "You must know this" or "Why would you.......". We live in a time anyone can access anything. I know old ladies and I am not making this up... that download the latest movies because they are tired of their TV provider. Does that mean they know coding? I mean come on. I know people that have programs from Github but have no idea how to download files from Github. Thanks to youtube videos and walkthroughs. We need to stop assuming and **try to help.** That was the WHOLE point of Github. 

Here are some additional points to consider:

1. **Installation Process Considerations:**
   - The statement "you're not supposed to do anything" oversimplifies the installation process and might be misleading for many users.

2. **Configuration Examination:**
   - When performing the installation, it's crucial to scrutinize the /config/extended.conf file to ensure that the setup aligns with your requirements.

3. **Understanding Folder Structures in Docker:**
   - The /config folder is located one level below the designated starting point. For example, if using Docker on a Synology and creating a Lidarr folder, the path would be /volume1/docker/lidarr/config. Users need to grasp the distinctions between Remote Path and Local Path, particularly when installing within Docker.

4. **SCRIPT ENABLEMENT Impact:**
   - In the SCRIPT ENABLEMENT section, the enableAutoConfig setting, when set to true, can significantly alter the Lidarr environment, impacting file naming structures and more. Users should comprehend this or consider starting Lidarr fresh with a separate installation.

5. **Managing Large Music Libraries:**
   - Lidarr, along with its scripts, may not be ideal for users with extensive music libraries. For those using Lidarr to upgrade music from MP3 to FLAC, setting up a separate environment with identical file naming and subsequently copying the files to the main library is recommended. Software like "Everything" can aid in this process. Use a search like this "D:\music\   ext:mp3    regex:^(.*)\.mp3$   sibling:"$1:.flac" In this example my MAIN music folder is D:\music Yours will more than likely be different.  Once the software does it's job you can safely delete the duplicate files since the naming structure are exactly the same. You will have then "upgraded" your music by removing the old duplicate song.

6. **Addressing Music Video Naming Issues:**
   - Users might encounter music videos with unconventional names like ~DIZ72.mp4. These are videos that had a "?" in the name. Utilizing tools such as Bulk Rename Utility, one can easily rectify this issue by referencing the real names within the accompanying .NFO files.

7. **Troubleshooting Tidal Integration:**
   - To enable Tidal functionality, closely monitor the logs while Lidarr is running (Just when it starts). For users employing a reverse proxy for containers (e.g., mydocker.mysite.com), ensuring proper IP address setup is crucial to avoid log viewing errors.

**Reply by WillScott73** [reply]
>  Utilizing tools such as Bulk Rename Utility, one can easily rectify this issue by referencing the real names within the accompanying .NFO files.

You can do that? 

Off I go to find out how....

**Reply by stuartmhannon** [reply]
@RandomNinjaAtk Can you address authenticating with tidal-dl? from what I can see in my install and reading the logs, tidal-dl is missing. 

root@c48d47e6f145:/custom-services.d# bash Audio
2023-11-26 12:56:50 :: Audio :: 2.23 :: Starting Script....
**/config/extended/functions: line 21: xq: command not found
/config/extended/functions: line 27: xq: command not found
/config/extended/functions: line 28: xq: command not found
/config/extended/functions: line 29: xq: command not found**
2023-11-26 12:56:50 :: Audio :: 2.23 :: -----------------------------------------------------------------------------
2023-11-26 12:56:50 :: Audio :: 2.23 ::  |~) _ ._ _| _ ._ _ |\ |o._ o _ |~|_|_|
2023-11-26 12:56:50 :: Audio :: 2.23 ::  |~\(_|| |(_|(_)| | || \||| |_|(_||~| | |<
2023-11-26 12:56:50 :: Audio :: 2.23 ::  Presents: Audio (2.23)
2023-11-26 12:56:50 :: Audio :: 2.23 ::  May the beats be with you!
2023-11-26 12:56:50 :: Audio :: 2.23 :: -----------------------------------------------------------------------------
2023-11-26 12:56:50 :: Audio :: 2.23 :: Donate: https://github.com/sponsors/RandomNinjaAtk
2023-11-26 12:56:50 :: Audio :: 2.23 :: Project: https://github.com/RandomNinjaAtk/arr-scripts
2023-11-26 12:56:50 :: Audio :: 2.23 :: Support: https://github.com/RandomNinjaAtk/arr-scripts/discussions
2023-11-26 12:56:50 :: Audio :: 2.23 :: -----------------------------------------------------------------------------
2023-11-26 12:56:55 :: Audio :: 2.23 :: 
2023-11-26 12:56:55 :: Audio :: 2.23 :: Lift off in...
2023-11-26 12:56:56 :: Audio :: 2.23 :: 5
2023-11-26 12:56:57 :: Audio :: 2.23 :: 4
2023-11-26 12:56:58 :: Audio :: 2.23 :: 3
2023-11-26 12:56:59 :: Audio :: 2.23 :: 2
2023-11-26 12:57:00 :: Audio :: 2.23 :: 1
2023-11-26 12:57:01 :: Audio :: 2.23 :: Add Deezer Top Artists is disabled (enable by setting addDeezerTopArtists=true)
2023-11-26 12:57:01 :: Audio :: 2.23 :: Add Deezer Top Album Artists is disabled (enable by setting addDeezerTopAlbumArtists=true)
2023-11-26 12:57:01 :: Audio :: 2.23 :: Add Deezer Top Track Artists is disabled (enable by setting addDeezerTopTrackArtists=true)
2023-11-26 12:57:01 :: Audio :: 2.23 :: Add Deezer Related Artists is disabled (enable by setting addRelatedArtists=true)
2023-11-26 12:57:01 :: Audio :: 2.23 :: Download Location: /config/extended/downloads/audio
2023-11-26 12:57:01 :: Audio :: 2.23 :: Output format: native
2023-11-26 12:57:01 :: Audio :: 2.23 :: Output bitrate: lossless
2023-11-26 12:57:01 :: Audio :: 2.23 :: Download Quality Check Enabled
2023-11-26 12:57:01 :: Audio :: 2.23 :: Preferred audio lyric type: both
2023-11-26 12:57:01 :: Audio :: 2.23 :: Tidal Country Code set to: US
2023-11-26 12:57:01 :: Audio :: 2.23 :: Replaygain Tagging Enabled
2023-11-26 12:57:01 :: Audio :: 2.23 :: Match Distance: 5
2023-11-26 12:57:01 :: Audio :: 2.23 :: Beets Tagging Enabled
2023-11-26 12:57:01 :: Audio :: 2.23 :: Beets Matching Threshold 90%
2023-11-26 12:57:01 :: Audio :: 2.23 :: Failed Download Attempt Theshold: 6
2023-11-26 12:57:01 :: Audio :: 2.23 :: TIDAL :: Verifying tidal-dl configuration
**2023-11-26 12:57:01 :: Audio :: 2.23 :: TIDAL :: No default config found, importing default config "tidal.json"
Audio: line 269: tidal-dl: command not found
2023-11-26 12:57:01 :: Audio :: 2.23 :: TIDAL :: ERROR :: Loading client for required authentication, please authenticate, then exit the client...
Audio: line 279: tidal-dl: command not found
2023-11-26 12:57:01 :: Audio :: 2.23 :: TIDAL :: Purging album list cache...
wget: bad port ''
wget: bad port ''
2023-11-26 12:57:01 :: Audio :: 2.23 :: FINDING MISSING ALBUMS :: sorted by date
2023-11-26 12:57:01 :: Audio :: 2.23 ::  Missing Albums Found!
2023-11-26 12:57:01 :: Audio :: 2.23 :: Getting Missing Album IDs
Audio: line 993: [: -ge: unary operator expected
wget: bad port ''**
2023-11-26 12:57:01 :: Audio :: 2.23 :: FINDING CUTOFF ALBUMS sorted by date
2023-11-26 12:57:01 :: Audio :: 2.23 ::  CutOff Albums Found Found!
2023-11-26 12:57:01 :: Audio :: 2.23 :: Getting CutOff Album IDs
Audio: line 1033: [: -ge: unary operator expected
2023-11-26 12:57:01 :: Audio :: 2.23 :: Script end...
2023-11-26 12:57:01 :: Audio :: 2.23 :: Script sleeping for 15m...

**Reply by KINGLIFER** [reply]
You need to attach to the console and see the url link the script provides. Then go to that link and authenticate. You wont find that in any documentation I found so hopefully that works for you.

**Reply by RandomNinjaAtk** [reply]
@stuartmhannon 

Your issue seems to be that you are missing the required packages (indicated by the error: command not found), which likely means the init script and the setup script did not run because those install all the required packages for everything to work.... So re-read the install/setup instructions and double check to make sure its correct or try again.... The scripts will never work if the requirements are not met...

**Reply by WillScott73** [reply]
Sounds like the issue I was having, whereby the setup scripts were not running. This was because I was cloning the repo to my Win 11 PC, and then copying to the linux box that my containers run on. 

The issue is with the EOL characters in the .sh files, not the script logic. Windows and Linux use different EOL characters.

**Reply by stuartmhannon** [reply]
@RandomNinjaAtk I ended up running the following from /custom-cont-init.d/:

curl https://raw.githubusercontent.com/RandomNinjaAtk/arr-scripts/main/lidarr/scripts_init.bash | bash

Then ran Audio service directly and was able to authenticate with Tidal.

**Comment by tikibozo** [comment]
I noticed my image pulls started failing and then saw the -extended containers archived and feared the worst until I found this repo! Great to see that you've just shifted your approach. Thanks for all your efforts!

**Comment by fradulentleech** [comment]
lidarr extended was nice but this is even better lol

**Comment by RandomUser345** [comment]
Happy to see this just moved into another direction, not being discontinued :) 
Trying to update to this solution in the next days: but it looks easy enough!

**Comment by espentruls** [comment]
I've had a problem with the new scripts for Lidarr. When I configure with an arl token it tells me that it is configured correctly, but when it tries to download it fails once and tries again and then fails and then restarts the entire audio script. The arl token worked with the old lidarr-extended docker container.


```07/13/2023 3:40:53 PM
2023-07-13 13:40:53 :: Audio :: 1.3 :: 1 :: missing :: 20 of 531 :: Ruth B. :: Safe Haven :: Album :: Download Attempt number 1
07/13/2023 3:40:54 PM
Paste here your arl:
07/13/2023 3:40:54 PM
Aborted!
07/13/2023 3:40:54 PM
2023-07-13 13:40:54 :: Audio :: 1.3 :: 1 :: missing :: 20 of 531 :: Ruth B. :: Safe Haven :: Album :: ERROR :: download failed, missing tracks...
07/13/2023 3:40:54 PM
2023-07-13 13:40:54 :: Audio :: 1.3 :: 1 :: missing :: 20 of 531 :: Ruth B. :: Safe Haven :: Album :: Retry Download in 1 second fix errors...
07/13/2023 3:40:55 PM
2023-07-13 13:40:55 :: Audio :: 1.3 :: 1 :: missing :: 20 of 531 :: Ruth B. :: Safe Haven :: Album :: Download Attempt number 2
07/13/2023 3:40:55 PM
Paste here your arl:
07/13/2023 3:40:55 PM
Aborted!
07/13/2023 3:40:55 PM
2023-07-13 13:40:55 :: Audio :: 1.3 :: DEEZER :: ERROR :: Download failed
07/13/2023 3:40:55 PM
2023-07-13 13:40:55 :: Audio :: 1.3 :: DEEZER :: ERROR :: Please review log for errors in client
07/13/2023 3:40:55 PM
2023-07-13 13:40:55 :: Audio :: 1.3 :: DEEZER :: ERROR :: Try updating your ARL Token to possibly resolve the issue...
07/13/2023 3:40:55 PM
2023-07-13 13:40:55 :: Audio :: 1.3 :: DEEZER :: ERROR :: Exiting...
```


I've tried using it without an arl token, but then Freyr uses youtube music to download extremely heavily compressed music. with lots of compression artifacts. So bad quality it's not even worth downloading.
Thought I should check to see if I'm the only one with this issue before opening a new issue.

**Reply by RandomNinjaAtk** [reply]
Make sure you've updated to the latest scripts:
https://github.com/RandomNinjaAtk/arr-scripts/tree/main/lidarr#updating

Also try updating your ARL token

**Reply by RandomNinjaAtk** [reply]
Still debugging it, if you already updated, update again...

**Reply by RandomNinjaAtk** [reply]
I can confirm the latest update fixed it. Give it a try

**Reply by espentruls** [reply]
The arl token was correct, you can easily verify it by using this easy python script. [stracc](https://rentry.co/stracc)
The latest update fixed it. 👍

**Comment by RandomUser345** [comment]
Happy to report that it works fine on arm64 without changes. Only using audio tough, no video.

**Reply by RandomNinjaAtk** [reply]
Thanks for the report, still working on minor bug fixes from the transition.

Maybe if people contact IMVDB, they'll fix their site? https://imvdb.com/about/contact

Keep in mind, you probably shouldn't reference this tool...

Edit:
I don't have any faith in them fixing it, it seems the site has been abandoned. I'm surprised it's still online...

**Comment by Castcore** [comment]
Thanks for these! I just have a question around what happens after you install the scripts. If I follow the steps: add the extra volume mappings, adding the "script_ini.bash", and then start the container. Will all the settings from extended.conf be applied immediately? I only really want to use the plex.notify script, but because I have my dockers setup slightly differently to TRaSH guides I don't want to accidentally overwrite all my settings with the auto config.
From my understanding it takes 2 restarts to full apply them, but then what happens on another restart when it runs "script_ini.bash" again will it overwrite the "extended.conf" file requiring the settings to be changed again? Sorry for my lack of understanding

**Reply by tikibozo** [reply]
I had a similar desire. If you grab the extended.conf from here and put it in config/extended.conf (as mapped via the volume), it won't be overwritten and you can customize it before you run the container with the scripts for the first time.

**Reply by Castcore** [reply]
I finally got around to installing this. Is there a way to enable only the Plex notify script inside the "extended.conf"? I'm not sure if I am missing something but I can't find an option for enabling the PlexNotify script inside the settings.

**Reply by Castcore** [reply]
I figured this one out actually. There are two ways if you ONLY want PlexNotifier.

1. AUTOMATIC: Configure "extended.conf", once you have provided the plex url and token you can set ' enableAutoConfig="true" ', then in the lower section "AUTOCONFIG SCRIPT SETTINGS" you can set all to false, except for ' configureCustomScripts="true" '. Then when you restart the docker it should set it up for you. In my case this worked for Radarr but not for Sonarr.
2. MANUAL: I would just use this method. Configure "extended.conf" with your plex url and token. Then go to the Sonarr/Radarr web gui, go to "Settings->Connect->+->Custom Script", and set the options as follows in both Radarr and Sonarr: 
![image](https://github.com/RandomNinjaAtk/arr-scripts/assets/88655407/10ea7afd-1c11-4989-8b0d-50b0cf47076a)

**Comment by WillScott73** [comment]
Would you be willing to add a .gitattributes file to the repo root with the content:

* text=auto
*.sh text eol=lf

It fixes the issue I raised a few days ago. I was cloning to Windows and then copying the files over to my linux server. Git was converting the lf to crlf and bash doesn't like that.

Thanks

**Reply by RandomNinjaAtk** [reply]
Make a PR for it ;)

**Comment by WillScott73** [comment]
Hopefully, I got that right. We use Gitlab at work, and it seems to be an easier workflow than this! I've never dared to contribute back to GitHub before.

**Reply by RandomNinjaAtk** [reply]
Welp, I guess you're going to find out if it works...

**Comment by Jdiesel87** [comment]
What ownership should the script be? I'm on unraid if that makes a difference. Trying to figure out if it should be root or nobody

**Reply by jaycedk** [reply]
In Unraid it needs to be the same permissions as the docker.
So nobody 😀

**Reply by RandomNinjaAtk** [reply]
I just download the script via CLI on unraid, as long as the container can read the file, that is all that is needed.

**Reply by KINGLIFER** [reply]
synology give an error. Says it needs to be root.

**Comment by pwrqwest** [comment]
Love the update. Is there a way to disable the Extras scan on docker start and only perform the scan at the internal time?

**Reply by RandomNinjaAtk** [reply]
No, there is no way to delay startups only time between runs.

**Comment by pwrqwest** [comment]
Is there a way to use Plexnotify via command line to have plex scan a single folder , Movie Title or TV show?

**Reply by RandomNinjaAtk** [reply]
It already does that...

**Reply by pwrqwest** [reply]
Thanks for responding. Can I manually kick off a scan outside of Sonarr and Radarr?

**Reply by RandomNinjaAtk** [reply]
No, you can’t easily for plexnotify because it requires data from sonarr to know what to scan. 

However, if you’re savvy, you could read through the script and use the correct variables/data from sonarr, then you can trigger the scan manually via cli when passing in the correct information possibly…. But it’s not something  it’s designed for and I’m going to support it.

**Comment by KINGLIFER** [comment]
I guess I am the only one that has no idea what to do here. Do I keep my old lidarr? must I be a past extended user? What if I have a configuration already setup in the arrs? What then?
What about video only?
It is so vague and 007ish in here I get nervous. LOL why is this so vague? I am not complaining more concerned about how to install.

**Reply by tikibozo** [reply]
If you follow the *arr instructions to set up one or all of them, then come here and follow the corresponding instructions to add these scripts, you'll be in good shape.

I was using the -extended containers before and was also confused initially at the shift here, but it turned out to be an easy migration once I understood what to do. If you have an existing installation, you can pull up your config file from it and move any customizations over to the new one. You can also do that before running this version for the first time (if you want/care.) Just grab the config file from this repo for the appropriate *arr, populate it, and drop it in the /config volume path before creating the new volumes/directories and bootstrap script (which are otherwise the only 2 setup steps iirc.)

You don't need to do anything in the *arr apps to set this up, the scripts will handle all that automatically.

**Reply by KINGLIFER** [reply]
Yup that did not work. It installs but not the scripts. Runs but just as lidarr. Nothing else.

**Comment by solidsnakex37** [comment]
Is it possible to define only certain extras to be downloaded, for the Radarr Extras script? 

For example, I can set all or trailers. If I do all, it can sometimes download 50+ extras, but in reality, all I want are: 

Trailers
Behind the Scenes

As an example. 

I assume it's only all or trailers, but wanted to check in case I was missing something. I'd love to only download specific options, not all or nothing

**Comment by botagas** [comment]
Hello,

Is it possible to use Extras only for a specific library? I am interested to exclude some libraries from downloading trailers.
